### PR TITLE
Reader: Improve stream headers

### PR DESCRIPTION
**Pull request by kean**
This PR updates the layout of the stream headers view to better work on iPad. It also lays the initial groundwork for supporting dynamically switching from compact to regular layout for future split view support.

Screenshot 2024-10-30 at 10 39 28 AM

To test:

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Reader/ReaderBaseHeaderView.swift [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func updateConstraints() {
+        NSLayoutConstraint.deactivate(contentViewConstraints)
+        contentViewConstraints = []
+
+        let insets = Self.makeInsets(isCompact: isCompact)
+        contentViewConstraints += contentView.pinEdges(.horizontal, to: isCompact ? self : readableContentGuide, insets: insets)
+        contentViewConstraints += contentView.pinEdges(.vertical, insets: insets)
```
Comment (crazytonyli) [inline WordPress/Classes/ViewRelated/Reader/ReaderBaseHeaderView.swift:33]: Nitpick: I believe the same thing can be implemented using `layoutMarginGuide`, which doesn't need a new "content view".

File: WordPress/Classes/ViewRelated/Reader/ReaderDiscoverHeaderView.swift [hunk 2]
```diff
@@ -22,10 +22,11 @@ final class ReaderDiscoverHeaderView: UIView, UITextViewDelegate {
         scrollView.clipsToBounds = false
         channelsStackView.pinEdges()
         scrollView.heightAnchor.constraint(equalTo: channelsStackView.heightAnchor).isActive = true
+        scrollView.contentInset = UIEdgeInsets(.leading, -10) // Align the "channels"
```
Comment (crazytonyli) [inline WordPress/Classes/ViewRelated/Reader/ReaderDiscoverHeaderView.swift]: I'm not sure how important it is to add these insets. Do the channels still align when increasing/decreasing the dynamic size?

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-iOS/milestone/297">25.5</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/Classes/ViewRelated/Reader/ReaderBaseHeaderView.swift (hunk 1 above)
Comment (kean) [reply WordPress/Classes/ViewRelated/Reader/ReaderBaseHeaderView.swift:33]: I'm also not entirely happy with the implementation and the complexity of manning `isCompact` and `readableContentGuide`, but I already tested it, and I'm running out of time for this project, so I have to keep it as is.

File: WordPress/Classes/ViewRelated/Reader/ReaderDiscoverHeaderView.swift (hunk 2 above)
Comment (kean) [reply WordPress/Classes/ViewRelated/Reader/ReaderDiscoverHeaderView.swift]: The margins in the channels don't scale with the dynamic type (for now), so it works OK. I actually totally forgot about the dynamic type, and I fixed it now and made a couple of more changes:

- Add support for Dynamic Type in Reader stream headers
- Fix an issue with background for channels in dark mode
- Fix an issue with Reader menu not using the large type

I also updated `scrollView.contentInset = UIEdgeInsets(.leading, -10)` to only work in non-compact mode. On iPhone, I think it looks better without this offset.

There is a known issue with Dynamic Type as the table view doesn't reload the header well when the dynamic size changes, but I think I'll just ignore it for now. I'm not sure how to best fix it, and it's a minor issues. Folks don't change dynamic type "dynamically".

**Review by crazytonyli — APPROVED** [review]

**Pull request merged**